src/impl/realm_coordinator.cpp

```diff
@@ -148,35 +147,52 @@ void RealmCoordinator::set_config(const Realm::Config& config)
 
 std::shared_ptr<Realm> RealmCoordinator::get_realm(Realm::Config config)
 {
-    std::lock_guard<std::mutex> lock(m_realm_mutex);
+    std::unique_lock<std::mutex> lock(m_realm_mutex);
 
     set_config(config);
 
+    auto schema = std::move(config.schema);
+    auto migration_function = std::move(config.migration_function);
```

Comment:
This is causing the test failure on Linux with GCC as it's clearing `config.migration_function`, preventing the test for `config.schema_mode == SchemaMode::ReadOnly && config.migration_function` within `Realm::open_with_config` from ever being true.